x-pack/legacy/plugins/reporting/export_types/csv_from_savedobject/server/execute_job.ts

```diff
@@ -45,7 +45,14 @@ function executeJobFactoryFn(server: ServerFacade): ExecuteJobFn {
     const jobLogger = logger.clone([jobId === null ? 'immediate' : jobId]);
 
     const { jobParams } = job;
-    const { isImmediate, panel, visType } = jobParams;
+    const { isImmediate, panel, visType } = jobParams as JobParamsPanelCsv & { panel: SearchPanel };
+
+    if (!panel) {
+      i18n.translate(
+        'xpack.reporting.exportTypes.csv_from_savedobject.executeJob.failedToAccessPanel',
+        { defaultMessage: 'Failed to access panel metadata for job execution' }
+      );
+    }
```

Comment:
This had to be added for types reasoning: `panel` can be undefined per the TS fixes in this PR.